markdown/events/2022/bytesize_nftest.md

```diff
@@ -22,3 +22,81 @@ It is our hope that these talks / videos will build an archive of training mater
+
+[12:57](https://www.youtube.com/watch?v=K9B7JRkMpQ4&t=777)
+Those are all great and lots of great things that you should go read about. But I think as a community, we have more important things to focus on. What are some realistic testing priorities? We should probably first convert our CI tests to pipeline tests. I think those are great to reproduce locally and quickly. Next, we can then add more pipeline tests for various params and pathways that you might want to make sure don't break, I think most people are just testing that all their liners work and then hope that everything downstream works, whereas you might want to check all of your skips, for example, and make sure that you get the right amount of processes and that certain things aren't getting run.
+
+[13:44](https://www.youtube.com/watch?v=K9B7JRkMpQ4&t=824)
+Another great practice that I hope that we can adopt is to add tests when fixing bugs in PRs that prove that this bug is fixed and we're not going to come back to this bug and checking for it in the future so that we don't end up recreating it. Lastly, I think testing your local modules can go a long way. I think we have a lot of custom scripts that we could do some tests on and make sure that we're getting the expected output out of those to verify our scientific results. So the TLDR is let's avoid regressions and move forward with our tests.
+
+[14:22](https://www.youtube.com/watch?v=K9B7JRkMpQ4&t=862)
+A couple of us have been toying around a lot how a roll out plan could look like. This is just a quick update on this and is no way indicative of exactly what's going to happen. First we want to start with pipelines as a proving ground because I think that's where we had the most ground to gain compared to being stuck with just GitHub actions. So it's already in methyl-seq and demultiplexed. If you're looking for some quick examples and you want to implement this today and switch those over, just that step one, we're going to need to update the template. I've got a PR started for that, but we need to settle on some best practices or how we want to do things so that as you switch between nf-core pipelines and contributing, things are familiar.
```

Comment:
```suggestion
A couple of us have been toying around a lot how a roll out plan could look like. This is just a quick update on this and is no way indicative of exactly what's going to happen. First, we want to start with pipelines as a proving ground because I think that's where we had the most ground to gain compared to being stuck with just GitHub actions. So it's already in methyl-seq and demultiplex. If you're looking for some quick examples and you want to implement this today and switch those over, just that step one, we're going to need to update the template. I've got a PR started for that, but we need to settle on some best practices or how we want to do things so that as you switch between nf-core pipelines and contributing, things are familiar.
```